Update generate-metadata.mdx (#81853)
vercel/next.js: docs/01-app/03-api-reference/04-functions/generate-metadata.mdx

> - You cannot export both the `metadata` object and `generateMetadata` function from the same route segment.
> - `fetch` requests inside `generateMetadata` are automatically [memoized](/docs/app/guides/caching#request-memoization) for the same data across `generateMetadata`, `generateStaticParams`, Layouts, Pages, and Server Components.
> - Resolving `generateMetadata` is part of rendering the page. If the page can be pre-rendered and `generateMetadata` doesn't introduce dynamic behavior, its result is included in the page’s initial HTML.

> > - `generateMetadata` results [can be streaming](/docs/app/api-reference/functions/generate-metadata#streaming-metadata) so it can serve the page a lot faster

Reviewer: ```suggestion
> - `generateMetadata` [can stream results](/docs/app/api-reference/functions/generate-metadata#streaming-metadata) to improve page load performance.
```

Author: @icyJoseph Yep I think that's more clear😁
